### PR TITLE
Reject line protocol that terminates with '-'

### PR DESCRIPTION
**Pull request by otoolep**
Fixes issue #4272.

### CODE REVIEW

**Comment by otoolep** [comment]
@jwilder

File: models/points.go [hunk 1]
```diff
@@ -560,6 +560,10 @@ func scanNumber(buf []byte, i int) (int, error) {
 	// Is negative number?
 	if i < len(buf) && buf[i] == '-' {
 		i += 1
+		// There must be more characters now.
```
Comment (jwilder) [inline models/points.go]: Can you add that this is checking for a field value of just '-'?

**Comment by jwilder** [comment]
Small comment suggestion :+1: though

File: models/points.go (hunk 1 above)
Comment (otoolep) [inline models/points.go]: Sure.

**Pull request merged**